
From: Martin Schwidefsky <schwidefsky@de.ibm.com>

sys_timer_settime returns EINVAL if *new is NULL, compat_timer_settime
returns EFAULT.  Add a check to compat_timer_settime to return EINVAL
if *new is NULL.


---

 25-akpm/kernel/compat.c |    3 +++
 1 files changed, 3 insertions(+)

diff -puN kernel/compat.c~s390-04-compat_timer_settime kernel/compat.c
--- 25/kernel/compat.c~s390-04-compat_timer_settime	Fri Feb 20 15:56:59 2004
+++ 25-akpm/kernel/compat.c	Fri Feb 20 15:56:59 2004
@@ -441,6 +441,9 @@ long compat_timer_settime(timer_t timer_
 	long err;
 	mm_segment_t oldfs;
 	struct itimerspec newts, oldts;
+
+	if (!new)
+		return -EINVAL;
 	if (get_compat_itimerspec(&newts, new))
 		return -EFAULT;	
 	oldfs = get_fs();

_
